### ISSUE TITLE
Nats healthcheck status unaivailable after deleting a JS account that still has ressources

### ISSUE THREAD

**Issue by amal-krm** [nats-io/nats-server#3685]
Hello,

When deleting an account that has JS resources (a stream with 3 replicas) in a cluster of 3 nodes, the healthcheck  of all the nodes of the servers are unavailable.
The cluster is still okay, we can create other streams, but the healthcheck is unavailable

 - [x] Included `nats-server -DV` [logs](https://github.com/nats-io/nats-server/files/10167359/log.tar.gz)
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve):
  ```
  nsc add account -n NatsTEST
  nsc add user -a NatsTEST -n test
  nsc edit account --name NatsTEST --js-mem-storage 1 --js-disk-storage 512M  --js-streams 10 --js-consumer 100
  nsc push -a NatsTEST
  nats --creds $NKEYS_PATH/creds/scw-dev-op/NatsTEST/test.creds --server 51.158.121.229 stream add
  nsc push -R NatsTEST
  ```

and for watching the healthcheck of servers:
```
watch curl 'http://cluster-mnq-1:8222/healthz'
watch curl 'http://cluster-mnq-2:8222/healthz'
watch curl 'http://cluster-mnq-3:8222/healthz'
```
#### Versions of `nats-server` and affected client libraries used:
  ```
    - nats_server_version: v2.9.7
    - nats_nsc_version: 2.7.3
    - nats cli version 0.0.35
  ```

#### OS/Container environment:

  `Ubuntu 20.04 (Focal Fossa)`

#### Steps or code to reproduce the issue:
    - create a cluster NATS with 3 nodes
    - watch curl of the healthcheck of a node
    - create an account
    - create user
    - Enable Jestream for the account
    - Create stream with replicas=3
    - Delete an account

#### Expected result:
healthcheck to be okay

#### Actual result:
healthcheck unavailable

**Comment by JulienVdG** [comment]
This looks very similar to  #3655 to me

**Comment by amal-krm** [comment]
Yes I understand, but here I'm deleting an account so it's supposed to not be enabled anymore.
according to your description it should not impact the healthcheck. (I know there are still some resource on disk but I'm not talking about this)

**Comment by derekcollison** [comment]
That is correct, if you purge the account startup should succeed and healthz report ok.

What is it reporting?

**Comment by derekcollison** [comment]
Our ops folks follow a process around this..

Generally we disable client connections to the account, we backup jetstream for the whole account, then delete (or js disable the account) then do account purge.
